lib/ProductOpener/Nutrition.pm

```diff
@@ -308,20 +373,1471 @@ sub convert_nutrient_to_100g ($nutrient_ref, $original_per, $original_per_quanti
+
+					my $input_set_ref = $input_sets_hash_ref->{$source}{$preparation}{$per};
+
+					remove_empty_nutrient_values_and_set_unspecified_nutrients($input_set_ref);
+
+					# Empty input sets are not stored
+					if (!exists $input_set_ref->{nutrients} or (keys %{$input_set_ref->{nutrients}}) == 0) {
+						next;
+					}
+
+					# Set the source, preparation and per as they may be only in the keys
+					# if we just created the input set from nutrient values
+					$input_set_ref->{source} = $source;
+					$input_set_ref->{preparation} = $preparation;
+					$input_set_ref->{per} = $per;
+
+					# Set the per quantity and unit for 100g, 100ml, 1l and 1kg
+					if ($per eq "100g") {
+						$input_set_ref->{per_quantity} = 100;
+						$input_set_ref->{per_unit} = "g";
+					}
+					elsif ($per eq "100ml") {
+						$input_set_ref->{per_quantity} = 100;
+						$input_set_ref->{per_unit} = "ml";
+					}
+					elsif ($per eq "1kg") {
+						$input_set_ref->{per_quantity} = 1000;
+						$input_set_ref->{per_unit} = "g";
+					}
+					elsif ($per eq "1l") {
+						$input_set_ref->{per_quantity} = 1000;
+						$input_set_ref->{per_unit} = "ml";
+					}
+					elsif ($per eq "serving") {
+						if (    (defined $product_ref->{serving_quantity})
+							and (defined $product_ref->{serving_quantity_unit}))
+						{
+							$input_set_ref->{per_quantity} = $product_ref->{serving_quantity};
+							$input_set_ref->{per_unit} = $product_ref->{serving_quantity_unit};
+						}
+					}
+
+					push(@{$input_sets_ref}, $input_set_ref);
+				}
+			}
+		}
+	}
+	return $input_sets_ref;
+}
+
+=head2 get_source_for_site_and_org ( $org_id = undef )
+
+Returns the default source of nutrition data for the current site and organization.
+
+=head3 Arguments
+
+=head4 $org_id
+
+Organization id
+
+=head3 Return values
+
+- "packaging" for the public platform
+- "manufacturer" for the pro platform
+
+=cut
+
+sub get_source_for_site_and_org ($org_id = undef) {
+
+	my $source = "packaging";
+	if ($server_options{producers_platform}) {
+		$source = "manufacturer";
+		if (defined $org_id) {
+			# e.g. org-database-usda
+			if ($org_id =~ /^org-database-(.+)$/) {
+				$source = "database-" . $1;
+			}
+			# e.g. org-label-gmo-project (in practice labels should not send nutrition data)
+			if ($org_id =~ /^org-label-(.+)$/) {
+				$source = "label-" . $1;
+			}
+			# At some point we used the pro platform to allow users to bulk enter data (e.g. for scan parties)
+			elsif ($org_id =~ /^user-(.+)$/) {
+				$source = "packaging";
+			}
+		}
+	}
+	return $source;
+}
+
+=head2 get_preparations_for_product_type
+
+Returns the list of valid preparation states for a given product type.
+
+=head3 Arguments
+
+=head4 $product_type
+
+Type of the product (food, petfood, etc)
+
+=head3 Return values
+
+List of valid preparation states for the given product type
+
+=cut
+
+sub get_preparations_for_product_type ($product_type) {
```

Comment:
It could also be a configuration value !

Reply:
I'll put it in Config.pm but keep the function to look it.